runtime/interpreter/encoding_test.go

```diff
@@ -0,0 +1,1938 @@
+package interpreter
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/cadence/runtime/ast"
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/sema"
+	"github.com/onflow/cadence/runtime/tests/utils"
+)
+
+type encodeDecodeTest struct {
+	value   Value
+	encoded []byte
+	invalid bool
+}
+
+func testEncodeDecode(t *testing.T, tests map[string]encodeDecodeTest) {
```

Comment:
It might be a good idea to repeat each test case in a loop (like a primitive fuzz test) to ensure that the encoding is fully deterministic, as done here: https://github.com/onflow/cadence/blob/master/encoding/xdr/encoding_test.go#L628-L633

Reply:
Great, idea, adding! 💯

Reply:
Added in 7444398